proxy-node-gateway/src/test/java/uk/gov/ida/notification/saml/validation/components/ComparisonValidatorTest.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.ida.notification.saml.validation.components;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+import org.opensaml.saml.saml2.core.AuthnContextComparisonTypeEnumeration;
+import org.opensaml.saml.saml2.core.RequestedAuthnContext;
+import org.opensaml.saml.saml2.core.impl.RequestedAuthnContextBuilder;
+
+import uk.gov.ida.notification.exceptions.authnrequest.InvalidAuthnRequestException;
+
+public class ComparisonValidatorTest {
+    private ComparisonValidator validator = new ComparisonValidator();
+
+    private static RequestedAuthnContext aRequestedAuthnContext(AuthnContextComparisonTypeEnumeration comparison) {
+        RequestedAuthnContext requestedAuthnContext = new RequestedAuthnContextBuilder().buildObject();
+        requestedAuthnContext.setComparison(comparison);
+        return requestedAuthnContext;
+    }
+
+    @Test
+    public void shouldThrowIfAuthnContextIsMissing() {
+        assertThrows(InvalidAuthnRequestException.class, () -> validator.validate(null));
```

Comment:
For some reason running the tests with gradle isn't picking this up, but with IntelliJ it is. `ComparisonValidator ` throws a `InvalidHubResponseException`, not a `InvalidAuthnRequestException ` when `authnContext` is missing I think?

Reply:
Same thing appears in Travis' logs for the build

Reply:
Wow, geez. We should fix that.

Reply:
Turns out tests in `ComparisonValidatorTest` weren't being run because they used the wrong annotation. This has been fixed.

Reply:
Interesting that the tests were run by IntelliJ - any idea why?